src/Library/demos/Banner/main.js

```diff
@@ -0,0 +1,42 @@
+import Gtk from "gi://Gtk?version=4.0";
+const Adw = imports.gi.Adw;
+Gtk.init();
+
+const _bannerBox = workbench.builder.get_object("banner_box");
+const _openBanner = workbench.builder.get_object("open_banner");
+
+function adwBanner() {
+  // Create a new AdwBanner widget
+  const banner = new Adw.Banner({
```

Comment:
Similar to https://github.com/sonnyp/Workbench/pull/248

please create the banner from the blueprint

Reply:
Since these are generally intended to be place in a window, directly under the header bar, it would be a good idea to create a small window to demonstrate this, too.

The button in Workbench could pop up a window, and contain a few widget to control the info bar's properties.

Reply:
Agree with Andy, the Banner should be in a Window.

No need for a custom button to open the window though - Workbench will directly offer the option if it detects a `Window` in the Blueprint. Similar to the "Window" library entry.

Reply:
Actually, the libadwaita demo doesn't use an external banner for its demo – I think it's safe to follow them and makes the demo visible immediately.

![image](https://user-images.githubusercontent.com/19673/228014592-5c1f49ba-ec4a-4104-a427-f23bb1e47443.png)

works for you @andyholmes ?

Reply:
Yes, that's good too. I think the important part is that it doesn't have margins on the top or left/right sides.

Replicating the libadwaita demo is good idea in general I think.